library/Icingadb/Model/Behavior/FlattenedObjectVars.php

```diff
@@ -66,6 +54,8 @@ public function set($name, $value)
             // the condition refer to a different column, which is totally irrelevant, but since it's always the same
             // column, the FilterProcessor won't attempt to combine the conditions. The literal icing on the cake.
             $condition->setColumn('always_the_same_but_totally_irrelevant');
+
+            return $condition;
```

Comment:
Since you're actually returning a *transformed* filter condition now, is the above workaround still required?
```php
$condition->setColumn('always_the_same_but_totally_irrelevant');
```

Reply:
Yes, the column name still doesn't change. The processor would then think that `host.vars.foo` and `host.vars.bar` are two different cols, although both translate `foo|bar` to `flatname=foo|flatname=bar`.

Reply:
> Yes, the column name still doesn't change.

However, this workaround is justified by not returning a transformed filter, which is no longer the case, and I see no difference between with and without this workaround, **both are broken**.

Filter: `service.vars.mixed_list[*]~bar|service.vars.mixed_list[*]~test`

**With "always_the_same_but_totally_irrelevant"**:
```sql
SELECT sub_customvar_flat_service.id AS sub_customvar_flat_service_id
FROM customvar_flat sub_customvar_flat
       INNER JOIN service_customvar sub_customvar_flat_service_customvar
                  ON sub_customvar_flat_service_customvar.customvar_id = sub_customvar_flat.customvar_id
       INNER JOIN service sub_customvar_flat_service
                  ON sub_customvar_flat_service.id = sub_customvar_flat_service_customvar.service_id
WHERE (sub_customvar_flat.flatname = 'mixed_list%5B%2A%5D')
  AND (sub_customvar_flat.flatvalue = 'bar|service.vars.mixed_list%5B%2A%5D~test')
```

**Without:**
```sql
SELECT sub_customvar_flat_service.id AS sub_customvar_flat_service_id
FROM customvar_flat sub_customvar_flat
       INNER JOIN service_customvar sub_customvar_flat_service_customvar
                  ON sub_customvar_flat_service_customvar.customvar_id = sub_customvar_flat.customvar_id
       INNER JOIN service sub_customvar_flat_service
                  ON sub_customvar_flat_service.id = sub_customvar_flat_service_customvar.service_id
WHERE (sub_customvar_flat.flatname = 'mixed_list%5B%2A%5D')
  AND (sub_customvar_flat.flatvalue = 'bar|service.vars.mixed_list%5B%2A%5D~test')
```

Reply:
Works for me. How do you test this?

Reply:
And: By *transforming* the change of the condition type is meant. Which still isn't the case.